transmart-rest-api/src/main/groovy/org/transmartproject/rest/serialization/HypercubeJsonSerializer.groovy

```diff
@@ -200,6 +159,52 @@ class HypercubeJsonSerializer extends HypercubeSerializer {
         cell
     }
 
+    protected void writeValue(Object value) {
+        if (value == null) {
+            writer.nullValue()
+        } else if (value instanceof String) {
+            writer.value(value as String)
+        } else if (value instanceof Number) {
+            writer.value(value as Double)
+        } else if (value instanceof Date) {
+            def time = Instant.ofEpochMilli((value as Date).time).toString()
+            writer.value(time)
+        } else if (value instanceof Map) {
+            def obj = value as Map
+            writer.beginObject()
+            obj.each { k, v ->
+                writer.name(k as String)
+                writeValue(v)
+            }
+            writer.endObject()
+        } else {
+            writer.value(value.toString())
+        }
+    }
+
+    protected void writeCell(Cell cell) {
+        writer.beginObject()
+        writer.name('inlineDimensions')
+        writer.beginArray()
+        cell.inlineDimensions.each {
+            writeValue(it)
+        }
+        writer.endArray()
+        writer.name('dimensionIndexes')
+        writer.beginArray()
+        cell.dimensionIndexes.each {
+            writer.value(it)
+        }
+        writer.endArray()
+        if (cell.numericValue != null) {
+            writer.name('numericValue').value(cell.numericValue)
+        } else if (cell.stringValue) {
+            writer.name('stringValue').value(cell.stringValue)
+        }
+        writer.endObject()
+        writer.flush()
```

Comment:
Why do you need to flush here?

Reply:
Not sure if it's needed. I don't know how the memory usage is when not explicitly calling flush. I guess it can go.